neqo-transport/src/quic_datagrams.rs

```diff
@@ -10,37 +10,61 @@ use crate::frame::{FRAME_TYPE_DATAGRAM, FRAME_TYPE_DATAGRAM_WITH_LEN};
 use crate::packet::PacketBuilder;
 use crate::recovery::RecoveryToken;
 use crate::stats::FrameStats;
-use crate::{ConnectionEvents, Error, Res};
+use crate::{events::OutgoingDatagramOutcome, ConnectionEvents, Error, Res};
 use neqo_common::Encoder;
 use std::cmp::min;
+use std::collections::VecDeque;
 use std::convert::TryFrom;
+use std::ops::Deref;
 
 pub const MAX_QUIC_DATAGRAM: u64 = 65535;
 
+struct QuicDatagram {
+    data: Vec<u8>,
+    id: Option<u64>,
```

Comment:
Why this `id` is optional? I think this should be always presented.

Reply:
We are not sure how useful it is and if it will be used, therefore it is optional.